### ISSUE TITLE
Exactly 4 minutes until SMA Home Manager is discovered

### ISSUE THREAD

**Issue by TheRealGros** [wimleers/homebridge-sma-home-manager#10]
**Paste of Logs:**
```

[17/03/2023, 14:18:23] [homebridge-sma-home-manager] This plugin is taking long time to load and preventing Homebridge from starting. See https://homebridge.io/w/JtMGR for more info.
[17/03/2023, 14:18:23] [homebridge-sma-home-manager] Discovered SMA inverter: { SerialNumber: 3010996510, FirmwareRevision: '1.1.48.R' }
[17/03/2023, 14:18:23] [homebridge-sma-home-manager] Discovered SMA energy manager: no
[17/03/2023, 14:18:24] [homebridge-sma-home-manager] Discovered SMA inverter: { SerialNumber: 3010996510, FirmwareRevision: '1.1.48.R' }
[17/03/2023, 14:18:24] [homebridge-sma-home-manager] Discovered SMA energy manager: no
[17/03/2023, 14:18:25] [homebridge-sma-home-manager] Discovered SMA inverter: { SerialNumber: 3010996510, FirmwareRevision: '1.1.48.R' }
[17/03/2023, 14:18:25] [homebridge-sma-home-manager] Discovered SMA energy manager: no
[17/03/2023, 14:18:26] [homebridge-sma-home-manager] Discovered SMA inverter: { SerialNumber: 3010996510, FirmwareRevision: '1.1.48.R' }
[17/03/2023, 14:18:26] [homebridge-sma-home-manager] Discovered SMA energy manager: no
[17/03/2023, 14:18:27] [homebridge-sma-home-manager] Discovered SMA inverter: { SerialNumber: 3010996510, FirmwareRevision: '1.1.48.R' }
[17/03/2023, 14:18:27] [homebridge-sma-home-manager] Discovered SMA energy manager: no
[17/03/2023, 14:18:28] [homebridge-sma-home-manager] Discovered SMA inverter: { SerialNumber: 3010996510, FirmwareRevision: '1.1.48.R' }
[17/03/2023, 14:18:28] [homebridge-sma-home-manager] Discovered SMA energy manager: no
[17/03/2023, 14:18:29] [homebridge-sma-home-manager] Discovered SMA inverter: { SerialNumber: 3010996510, FirmwareRevision: '1.1.48.R' }
[17/03/2023, 14:18:29] [homebridge-sma-home-manager] Discovered SMA energy manager: no
[17/03/2023, 14:18:30] [homebridge-sma-home-manager] Discovered SMA inverter: { SerialNumber: 3010996510, FirmwareRevision: '1.1.48.R' }
[17/03/2023, 14:18:30] [homebridge-sma-home-manager] Discovered SMA energy manager: no

```

**Config:**

```json


```

**Screenshots:**

**Inverter:**

* **Model:** 

**Homebridge Environment:**

* **Node.js Version**: 
* **NPM Version**: 
* **Homebridge Version**: 
* **homebridge-sma-inverter Version**: 
* **Operating System**: Raspbian / Ubuntu / Debian / Windows / macOS / Docker / other

**Comment by wimleers** [comment]
This means that the logic that was introduced to solve #3 causes problems for at least *some* users 😔

But … I’m sure that once we hear back from @TheRealGros, I’ll be able to figure out a way to make this work 😊

**Comment by wimleers** [comment]
Woah! It takes at most a few seconds here!

Maybe the SMA Home Manager was just restarting?

If you restart Homebridge, does it happen again?

**Comment by wimleers** [comment]
Either way, I’ll tweak the logging so it’s less noisy.

**Comment by wimleers** [comment]
Woah! Bizarre. Could you please describe your home networking setup? And specifically the path network packets have to travel to go from Homebridge to your Home Manager.

**Comment by wimleers** [comment]
Did some research on this and failed to find anything related to zeroconf/Bonjour/mDNS/DNS-SD discovery latency.

Then I took another look at your logs and  noticed … that it's not actually the inverter that is taking 4 minutes to be found, it's your SMA Home Manager! 😅

That mechanism has not changed _at all_ since `1.0.0`. That means that this MUST be a completely independent problem. SMA's Home Manager 2.0 is _always_ broadcasting its information using multicast, (specifically [IGMP v3](https://en.wikipedia.org/wiki/Internet_Group_Management_Protocol).

We're just telling the operating system that we are also interested in it. But it also requires those packages to travel freely between your SMA Home Manager 2.0 and the device that your Homebridge instance is running on. If that's not happening, then there likely is a switch, router or access point on the network path between them that does not support/is not correctly configured for IGMP.

**Comment by wimleers** [comment]
Needed to improve documentation:
1. OS, OS version, `node` version for the device Homebridge is running on
2. network path between inverter & SMA Home Manager
3. network path between SMA Home Manager & Homebridge

**Comment by wimleers** [comment]
Bump 😊

**Comment by TheRealGros** [comment]
:-)Finally made it. I have seen that I had issues with ipv6, so I disabled it completely on my pi.Then I updated to plugin v1.1.3, this is the output:26/04/2023, 16:19:07] [homebridge-sma-home-manager] Launched child bridge with PID 2815[26/04/2023, 16:19:07] Registering platform 'homebridge-sma-home-manager.SMAHomeManager'[26/04/2023, 16:19:07] [homebridge-sma-home-manager] Loaded homebridge-sma-home-manager v1.1.3 child bridge successfully[26/04/2023, 16:19:08] [homebridge-sma-home-manager] Found inverter SMA3010996510 at SMA3010996510.local with IPv4 address 192.168.188.85.[26/04/2023, 16:19:08] [homebridge-sma-home-manager] Discovered SMA inverter: no[26/04/2023, 16:19:08] [homebridge-sma-home-manager] Discovered SMA energy manager: { Model: 372, SerialNumber: 3011955078, FirmwareRevision: '2.11.5.R' }[26/04/2023, 16:19:09] [homebridge-sma-home-manager] Discovered SMA inverter: {  DeviceClass: 'solar inverter',  DeviceType: 9348,  SerialNumber: 3010996510,  FirmwareRevision: 'unknown'}[26/04/2023, 16:19:09] [homebridge-sma-home-manager] Discovered SMA energy manager: { Model: 372, SerialNumber: 3011955078, FirmwareRevision: '2.11.5.R' }[26/04/2023, 16:19:09] Initializing platform accessory 'Live power flow'...[26/04/2023, 16:19:09] Initializing platform accessory 'Recent power flow'...[26/04/2023, 16:19:09] Initializing platform accessory 'Today's energy flow'...[26/04/2023, 16:19:09] Initializing platform accessory 'Off Grid energy signal'...[26/04/2023, 16:19:09] Initializing platform accessory 'No Sun energy signal'...[26/04/2023, 16:19:10] Homebridge v1.6.0 (HAP v0.11.0) (homebridge-sma-home-manager) is running on port 49684.—————————————————-Gesendet von ***@***.*** 26.04.2023 um 14:22 schrieb Wim Leers ***@***.***>:﻿
Bump 😊

**Comment by TheRealGros** [comment]
Yes!-----------Am 26.04.2023 um 19:02 schrieb Wim Leers ***@***.***>:﻿
Looks like it worked within seconds this time around? 😊
Is everything working fine for you now?

**Comment by wimleers** [comment]
Glad to hear it 😊

Hope you enjoy using my plugin!

**Comment by TheRealGros** [comment]
Since today….—————————————————-Gesendet von ***@***.*** 26.04.2023 um 21:33 schrieb Wim Leers ***@***.***>:﻿
Closed #10 as completed.
